FWCore/Framework/test/test_module_delete_subprocess_cfg.py

```diff
@@ -168,7 +168,10 @@ def nonEventConsumer(transition, sourcePattern, expected):
 subprocessB.consumerEventFromA = intEventConsumer.clone(moduleLabel = "producerAEventConsumedInB", valueMustMatch = 1)
 subprocessB.consumerBeginLumiFromA = nonEventConsumer("beginLumi", "producerA%sConsumedInB", 2)
 subprocessB.consumerEndRunFromA = nonEventConsumer("endRun", "producerA%sConsumedInB", 5)
-subprocessB.consumerBeginProcessBlockFromA = nonEventConsumer("beginProcessBlock", "producerA%sConsumedInB", 6)
+subprocessB.consumerBeginProcessBlockFromA = intNonEventProducer.clone(
+    consumesAccessInputProcessBlock = cms.InputTag("producerABeginProcessBlockConsumedInB", "beginProcessBlock"),
+    expectAccessInputProcessBlock = cms.untracked.int32(6)
+)
```

Comment:
Just to check that I understood. The SubProcess uses the accessInputProcessBlock to access ProcessBlock products of the parent Process, as if the ProcessBlock products would be in an input file, correct?

I tried to look for an answer from the twiki page, but did not quickly find a crystal clear answer. I did became a bit confused by
> [on SubProcess] One can get ProcessBlock products from the input file during AccessInputProcessBlock transitions, but one cannot access other products associated with an ancestor process during Event, LuminosityBlock or Run processing. The AccessInputProcessBlock transition will run, but not until the endProcessBlock transition has completed. With the current design of SubProcess, this will occur after Event, LuminosityBlock and Run processing is complete.

in https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideProcessBlockData#Limitations_and_Details

Does it mean that the accessInputProcessBlock transition of a SubProcess will run only after the Event+Lumi+Run processing of the parent process is complete?

Reply:
Short answer: Yes.

I deleted the paragraph you referenced and added a longer entire section related to SubProcess. Hopefully, this is more clear. And I also added a new section on the page with the SubProcess documentation that you and/or Chris might want to review...

https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideProcessBlockData#SubProcess_and_ProcessBlock
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideSubProcesses#A_Problem

Reply:
Thanks, the documentation describes the behavior and the issue well now.